idx/idx.go

```diff
@@ -29,6 +35,42 @@ type Archive struct {
 	LastSave uint32 // last time the metricDefinition was saved to a backend store (cassandra)
 }
 
+type MetricID struct {
+	org int
+	key [16]byte
+}
+
+func NewMetricIDFromString(s string) (MetricID, error) {
+	id := MetricID{}
+	err := id.FromString(s)
+	return id, err
+}
+
+func (id *MetricID) FromString(s string) error {
+	splits := strings.Split(s, ".")
+	if len(splits) != 2 || len(splits[1]) != 32 {
+		return errInvalidIdString
+	}
+
+	var err error
+	id.org, err = strconv.Atoi(splits[0])
+	if err != nil {
+		return err
+	}
+
+	dst := make([]byte, 16)
+	n, err := hex.Decode(dst, []byte(splits[1]))
+	if n != 16 {
+		return errInvalidIdString
+	}
+	copy(id.key[:], dst)
+	return nil
+}
+
+func (id *MetricID) String() string {
+	return fmt.Sprintf("%d.%032x", id.org, id.key)
```

Comment:
with the code and bench i posted earlier, using `%032x` seems to cause a 20~30ns slowdown as opposed to just `%x`

Reply:
ok, seems you're right. I'm surprised that this prints the leading `0`s.